guide.js

```diff
@@ -363,47 +351,53 @@ function updateIframe(visible){
 	}
 	else{
 		//iframe going from on to off
-		//just hide it because if we go back to large, we don't want to have to reload the iframe.
-		let sidebar= document.getElementById("sidebarContent");
-		if(sidebar != null){
+		//just hide the entire side panel because if we go back to large, we don't want to have to reload the iframe.
+		let sidebar = document.getElementById("sidebar");
+		let divider = document.getElementById("dragMe");
+		let mainPanel = document.getElementsByClassName("mainPanel")[0]; //TODO: homogenize the id for this div on each webpage. //DOES THIS GRAB OTHER GUIDE FRAMES FROM WITHIN THE IFRAME? LOWER SCOPE TO A GUIDE ID?
+		if(sidebar != null && divider != null && mainPanel != null){
 			sidebar.style.display = "none";
+			divider.style.display = "none";
+			mainPanel.style.width = "100%"; //Want this to be 100% width since draggable will set it assuming there is a sidePanel even if there isn't.
 		}
-
-		if(settings.iframeCheck == "window"){
-			var links = document.getElementsByTagName("A");
+		//update links to redirect to desired place based on settings.
+		if(settings.iframeCheck == "window"){ //If the user wants links to redirect to a second window.
+			var links = document.getElementsByClassName("mainPanel")[0].getElementsByTagName("A"); //DOES THIS GRAB OTHER GUIDE FRAMES FROM WITHIN THE IFRAME? LOWER SCOPE TO A GUIDE ID?
 			for(var lnk of links){
-				if(lnk.target == "_blank"){
-					lnk.target = "myframe";
+				if(lnk.target == "_blank" || lnk.target == "myframe"){
+					lnk.target = "externalSecondWindow";
 				}
 			}
 		}
-		else{
-			//reset links to open in new tab, otherwise it looks like they're doing nothing.
-			var links = document.getElementsByTagName("A");
+		else{//iframeCheck setting is "off" and the user doesn't specfically want it to open in a second window, so we do new tab.
+			var links = document.getElementsByClassName("mainPanel")[0].getElementsByTagName("A"); //DOES THIS GRAB OTHER GUIDE FRAMES FROM WITHIN THE IFRAME? LOWER SCOPE TO A GUIDE ID?
 			for(var lnk of links){
-				if(lnk.target == "myframe"){
+				if(lnk.target == "myframe" || lnk.target == "externalSecondWindow"){
 					lnk.target = "_blank";
 				}
 			}
 		}
 		__displayingIframe = false;
 	}
+	__linkFormatType = settings.iframeCheck;
 }
 
+//The root function of the iframe resizing settings, runs each time the window size changes.
 var windowResizeId = null;
 function onWindowResize(event){
 	//on window resize, we may want to hide sidebar.
-	//only resize after being still for 50ms
+	//only resize after being still for 10ms
```

Comment:
why the time decrease?

Reply:
50ms was actually noticeable, to the extent that I initially thought my code was somehow being really slow to recalculate positions and stuff. Idk if my high refresh rate monitor is contributing to my perceived lag for 50ms, but at 10ms I didn't see any performance issues and it just looked more smooth and immediate. I imagine an end user will be confused by the lag too so I left the change in.